### DISCUSSION TITLE
Can poetry support multiple packages?

### DISCUSSION THREAD

**Discussion by Sigurdur42** [python-poetry/poetry#9489] [Q&A] [answered]
I currently have a larger application which I'd like to split into multiple packages but within a single pyproject.toml.
So, the structure would look like
`
- pyproject.toml
- package_ui
  - pyhton files
- package_logic
  - pyhton files
`

Is this supported? I want to use the packages only for beter structuring my large project.
Thx

**Comment by mgvalverde** [comment, marked as answer]
Hi, 

yes it can. 

You'd need to add in your `pyproject.toml` a block like this: 
```toml
[tool.poetry]
....
packages = [
    {include = "<pkg_folder_1>", from="..."}, 
    {include = "<pkg_folder_N>", from="..."},
]
```

I pushed a demo repo [here](https://github.com/mgvalverde/poetry_multipackage_example), in case you want to have a look.

**Reply by Sigurdur42** [reply]
Thanks, that worked.

**Reply by puc9** [reply]
What an answer, thanks a lot!

I could not figure out why suddenly poetry stopped adding `src` to the end of the `project.pth` file in the virtual environment.

I was using:

```toml

[tool.poetry]
....
packages = [
    {include = "src/<pkg_folder_1>"}, 
    {include = "src/<pkg_folder_N>"},
]

```
Adding that `from="src"` and removing `src/` from `include` fixed it.
